### PR TITLE
Move Breeze documentation to inside Breeze and split it.

### PR DESCRIPTION
**Pull request by potiuk**
The BREEZE.rst document became enormous - enough for it to stop being rendered by GitHub. This change splits it into multiple smaller documents - each focusing on a specific aspect of Breeze, making it possible by the user to focus on only that aspect that the user is interested at.

We also add nice index that guides the user to know about all the aspects of Breeze.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by eladkal — APPROVED** [review]
Nice!

File: BREEZE.rst [hunk 1]
```diff
@@ -15,2895 +15,4 @@
...
-
-    <div align="center">
-        <img src="images/pycharm_debug_breeze.png" width="640"
-             alt="Airflow Breeze - PyCharm debugger configuration">
-    </div>
-
-Then you can setup breakpoints and debug breeze as any other Python script or test.
-
-Similar configuration can be done for VSCode.
+Content of this guide have been moved to `Breeze docs internal folder <dev/breeze/doc/BREEZE.rst>`_
```
Comment (pankajkoti) [inline BREEZE.rst]:
```suggestion
Content of this guide has been moved to `Breeze docs internal folder <dev/breeze/doc/BREEZE.rst>`_
```

**Review by pankajkoti — APPROVED** [review]

File: BREEZE.rst (hunk 1 above)
Comment (potiuk) [reply BREEZE.rst]: Yep. solved.

**Comment by ferruzzi** [comment]
That was a big one, thanks for that.

**Pull request merged**